File: Paladins.Api/Paladins.Api/Paladins.Common/Models/ChampionModel.cs
﻿using Newtonsoft.Json;
using Paladins.Common.Interfaces.Definitions;
using System;
using System.Collections.Generic;

namespace Paladins.Common.Models
{
    public class ChampionModel : IModel
    {
        public ChampionModel()
        {
            Abilities = new List<AbilityModel>();
            ChampionLinkageModels = new List<ChampionLinkageModel>();
        }
        public int Id { get; set; }
        public int PaladinsChampionId { get; set; }
        public string LatestChampion { get; set; }
        public string Role { get; set; }
        public int Speed { get; set; }
        public string Title { get; set; }
        public string Name { get; set; }
        public string Lore { get; set; }
        public int Health { get; set; }
        public Uri ChampionIconUrl { get; set; }
        public IEnumerable<AbilityModel> Abilities { get; set; }

        [JsonIgnoreAttribute]
        public List<ChampionLinkageModel> ChampionLinkageModels { get; set; }
    }

    
}

File: Mobile/app/src/main/java/com/fivegearszerochill/paladins/domain/models/initial/ChampionModel.kt
package com.fivegearszerochill.paladins.domain.models.initial

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

data class ChampionModel (
    @Expose
    @SerializedName("id")
    val id: Long,
    @Expose
    @SerializedName("paladinsChampionId")
    val paladinsChampionID: Long,
    @Expose
    @SerializedName("latestChampion")
    val latestChampion: String,
    @Expose
    @SerializedName("role")
    val role: String,
    @Expose
    @SerializedName("speed")
    val speed: Long,
    @Expose
    @SerializedName("title")
    val title: String,
    @Expose
    @SerializedName("name")
    val name: String,
    @Expose
    @SerializedName("lore")
    val lore: String,
    @Expose
    @SerializedName("health")
    val health: Long,
    @Expose
    @SerializedName("championIconUrl")
    val championIconURL: String,
    @Expose
    @SerializedName("abilities")
    val abilities: List<ChampionAbilityModel>
)